integration-tests/stateful-pallet-storage/handleAppendOnly.test.ts

```diff
@@ -0,0 +1,76 @@
+import assert from "assert";
+import {
+  createDelegatorAndDelegation,
+  createProviderKeysAndId,
+  getCurrentItemizedHash,
+} from "../scaffolding/helpers";
+import { KeyringPair } from "@polkadot/keyring/types";
+import { ExtrinsicHelper } from "../scaffolding/extrinsicHelpers";
+import { AVRO_CHAT_MESSAGE } from "./fixtures/itemizedSchemaType";
+import { MessageSourceId, SchemaId } from "@frequency-chain/api-augment/interfaces";
+import {Bytes, u16} from "@polkadot/types";
+
+describe("📗 Stateful Pallet Storage AppendOnly Schemas", () => {
+    let itemizedSchemaId: SchemaId;
+    let msa_id: MessageSourceId;
+    let providerId: MessageSourceId;
+    let providerKeys: KeyringPair;
+
+    before(async function () {
+
+        // Create a provider for the MSA, the provider will be used to grant delegation
+        [providerKeys, providerId] = await createProviderKeysAndId();
+        assert.notEqual(providerId, undefined, "setup should populate providerId");
+        assert.notEqual(providerKeys, undefined, "setup should populate providerKeys");
+
+        // Create a schema for Itemized PayloadLocation
+        const createSchema = ExtrinsicHelper.createSchemaWithSettings(providerKeys, AVRO_CHAT_MESSAGE, "AvroBinary", "Itemized", "AppendOnly");
+        const [event] = await createSchema.fundAndSend();
+        if (event && createSchema.api.events.schemas.SchemaCreated.is(event)) {
+            itemizedSchemaId = event.data.schemaId;
+        }
+        assert.notEqual(itemizedSchemaId, undefined, "setup should populate schemaId");
+
+        // Create a MSA for the delegator and delegate to the provider for the itemized schema
+        [, msa_id] = await createDelegatorAndDelegation(itemizedSchemaId, providerId, providerKeys);
+        assert.notEqual(msa_id, undefined, "setup should populate msa_id");
+    });
+
+    describe("Itemized With AppendOnly Storage Tests", () => {
+
+      it("should not be able to call delete action", async function () {
+
+        // Add and update actions
+        let payload_1 = new Bytes(ExtrinsicHelper.api.registry, "Hello World From Frequency");
+
+        const add_action = {
+          "Add": payload_1
+        }
+
+        let payload_2 = new Bytes(ExtrinsicHelper.api.registry, "Hello World Again From Frequency");
+
+        const update_action = {
+          "Add": payload_2
+        }
+
+        const idx_1: u16 = new u16(ExtrinsicHelper.api.registry, 1)
+
+        const delete_action = {
+          "Delete": idx_1
+        }
+        const target_hash = await getCurrentItemizedHash(msa_id, itemizedSchemaId);
+
+        let add_actions = [add_action, update_action, delete_action];
+
+        let itemized_add_result_1 = ExtrinsicHelper.applyItemActions(providerKeys, itemizedSchemaId, msa_id, add_actions, target_hash);
+        await assert.rejects(async () => {
+          await itemized_add_result_1.fundAndSend();
+        }, {
+          name: 'SchemaNotSupported',
```

Comment:
This returned error doesn't make sense to me in this case. I would've expected this to return `InvalidItemAction` instead. I see in the pallet code that's what it returns, it's just surprising.

Reply:
We're sending an action array that contains non-`Add` actions to an `AppendOnly` schema. The `check_schema` function returns `SchemaNotSupported`.

Reply:
Thanks @aramikm, it makes sense, I was coming at it from the standpoint of someone looking at their error, and there are a couple of places where that error crops up so I was thinking it could be good to have something more distinct in this case.

Reply:
Changed it to **UnsupportedOperationForSchema** if you think it still sucks please let me know what would you name it